cmd/print.go

```diff
@@ -43,55 +46,139 @@ var (
+	namespaceFilter string
+}
+
+// PrintGatewaysAndHttpRoutes performs necessary steps to digest and print
+// converted Gateways and HTTP Routes. The steps includes reading from the source,
+// construct ingresses, convert them, then print them out.
+func (pr *PrintRunner) PrintGatewaysAndHttpRoutes(cmd *cobra.Command, args []string) error {
+	err := pr.initializeResourcePrinter()
+	if err != nil {
+		fmt.Printf("Failed to initialize resrouce printer: %v", err)
+		os.Exit(1)
+	}
+	err = pr.initializeNamespaceFilter()
+	if err != nil {
+		fmt.Printf("Failed to initialize namespace filter: %v", err)
+		os.Exit(1)
+	}
+
+	ds := datasource.DataSource{
+		NamespaceFilter: pr.namespaceFilter,
+		InputFile:       pr.inputFile,
+	}
+	ingressList, err := ds.GetIngessList()
+	if err != nil {
+		fmt.Printf("Failed to get ingresses from source: %v", err)
+		os.Exit(1)
+	}
+
+	httpRoutes, gateways, errors := i2gw.Ingresses2GatewaysAndHTTPRoutes(ingressList.Items)
+	if len(errors) > 0 {
+		fmt.Printf("# Encountered %d errors\n", len(errors))
+		for _, err := range errors {
+			fmt.Printf("# %s\n", err)
+		}
+		os.Exit(1)
+	}
+
+	pr.outputResult(httpRoutes, gateways)
+
+	return nil
+}
+
+func (pr *PrintRunner) outputResult(httpRoutes []gatewayv1beta1.HTTPRoute, gateways []gatewayv1beta1.Gateway) {
+	for i := range gateways {
+		err := pr.resourcePrinter.PrintObj(&gateways[i], os.Stdout)
 		if err != nil {
-			return err
+			fmt.Printf("# Error printing %s HTTPRoute: %v\n", gateways[i].Name, err)
 		}
-		namespaceFilter, err := getNamespaceFilter(namespace, allNamespaces)
+	}
+
+	for i := range httpRoutes {
+		err := pr.resourcePrinter.PrintObj(&httpRoutes[i], os.Stdout)
 		if err != nil {
-			return err
+			fmt.Printf("# Error printing %s HTTPRoute: %v\n", httpRoutes[i].Name, err)
 		}
-		i2gw.Run(resourcePrinter, namespaceFilter, inputFile)
-		return nil
-	},
+	}
 }
 
-// getResourcePrinter returns a specific type of printers.ResourcePrinter
-// based on the provided outputFormat.
-func getResourcePrinter(outputFormat string) (printers.ResourcePrinter, error) {
-	switch outputFormat {
+// initializeResourcePrinter assign a specific type of printers.ResourcePrinter
+// based on the outputFormat of the printRunner struct.
+func (pr *PrintRunner) initializeResourcePrinter() error {
+	switch pr.outputFormat {
 	case "yaml", "":
-		return &printers.YAMLPrinter{}, nil
+		pr.resourcePrinter = &printers.YAMLPrinter{}
+		return nil
 	case "json":
-		return &printers.JSONPrinter{}, nil
+		pr.resourcePrinter = &printers.JSONPrinter{}
+		return nil
 	default:
-		return nil, fmt.Errorf("%s is not a supported output format", outputFormat)
+		return fmt.Errorf("%s is not a supported output format", pr.outputFormat)
 	}
+
 }
 
-// getNamespaceFilter returns a namespace filter, taking into consideration whether a specific
+// initializeNamespaceFilter generate the corret namespace filter, taking into consideration whether a specific
```

Comment:
Can we add a nice longer comment here about the actual logic. It is not straight forward so a comment could be nice here